web/packages/teleterm/src/ui/ConnectMyComputer/DocumentConnectMyComputerSetup/DocumentConnectMyComputerSetup.story.tsx

```diff
@@ -16,12 +16,52 @@
 
 import React from 'react';
 
+import { MockAppContextProvider } from 'teleterm/ui/fixtures/MockAppContextProvider';
+import { MockAppContext } from 'teleterm/ui/fixtures/mocks';
+import { WorkspaceContextProvider } from 'teleterm/ui/Documents';
+import * as types from 'teleterm/ui/services/workspacesService';
+
+import { makeRootCluster } from 'teleterm/services/tshd/testHelpers';
+
 import { DocumentConnectMyComputerSetup } from './DocumentConnectMyComputerSetup';
 
 export default {
   title: 'Teleterm/ConnectMyComputer/DocumentConnectMyComputerSetup',
 };
 
 export function Default() {
-  return <DocumentConnectMyComputerSetup visible={true} doc={undefined} />;
+  const cluster = makeRootCluster();
+  const doc: types.DocumentConnectMyComputerSetup = {
+    kind: 'doc.connect_my_computer_setup',
+    rootClusterUri: cluster.uri,
+    title: 'Connect My Computer',
+    uri: '/docs/123',
+  };
+  const appContext = new MockAppContext();
+  appContext.clustersService.state.clusters.set(cluster.uri, cluster);
+  appContext.workspacesService.setState(draftState => {
+    draftState.rootClusterUri = cluster.uri;
+    draftState.workspaces[cluster.uri] = {
+      localClusterUri: cluster.uri,
+      documents: [doc],
+      location: doc.uri,
+      accessRequests: undefined,
+    };
+  });
+
+  return (
+    <MockAppContextProvider appContext={appContext}>
+      <WorkspaceContextProvider
```

Comment:
The ergonomics of setting up a mock of this context in a story/tests are so bad. 🫠

I think we should have `MockWorkspaceContextProvider` that receives `rootClusterUri` and sets everything up from the context.

Reply:
Actually, it could work this way in production code. 
We would pass only `rootClusterUri` to `WorkspaceContextProvider` and it could read all necessary properties from the `AppContext`. 

If you have some spare time you can change it, or we can can just leave a TODO :)

Reply:
In production code, it's probably better to just call it the way we call it now since we have everything available from the get go when iterating over workspaces.

OTOH, I don't understand how the current code works. `DocumentsRenderer` is rendered without any props and doesn't subscribe to any updates, so I don't know how it gets rerendered during runtime and when it passes different values to `WorkspaceContextProvider`.

`useWorkspaceContext` calls `ctx.workspacesService.useState()` which just seems to trigger a rerender in any component that uses `useWorkspaceContext` when `workspacesService` state changes.

But I don't understand how components can actually observe different values being passed through the workspace context. 🤔

Reply:
We have `ctx.workspacesService.useState()` in `TabHostContainer` that triggers a rerender of everything below.